programs/staking/src/vault.rs

```diff
@@ -90,10 +90,18 @@ impl Vault {
 
     /// Stake token into the vault, and return share amount (LP)
     pub fn stake(&mut self, current_time: u64, token_amount: u64, lp_supply: u64) -> Option<u64> {
+        // vault is empty, guarantee that lp_supply is 0
         if self.total_amount == 0 {
```

Comment:
Do you think we can remove this? because the case `lp_supply == 0` has covered it
Never having the case that `self.total_amount ==.0 && lp_supply != 0`

Reply:
Fixed